nikola/plugin_categories.py

```diff
@@ -451,3 +452,141 @@ def import_file(self):
+
+    def get_list_path(self, lang):
+        """A path handler for the list of all classifications.
+
+        The last element in the returned path must have no extension, and the
+        PRETTY_URLS config must be ignored. The return value will be modified
+        based on the PRETTY_URLS and INDEX_FILE settings.
+        """
+        raise NotImplementedError()
+
+    def get_path(self, classification, lang):
+        """A path handler for the given classification.
+
+        The last element in the returned path must have no extension, and the
+        PRETTY_URLS config must be ignored. The return value will be modified
+        based on the PRETTY_URLS and INDEX_FILE settings.
+
+        For hierarchical taxonomies, the result of extract_hierarchy is provided.
+        For non-hierarchical taxonomies, the classification string itself is provided.
+        """
+        raise NotImplementedError()
+
+    def extract_hierarchy(self, classification):
+        """Given a classification, return a list of parts in the hierarchy.
+
+        For non-hierarchical taxonomies, it usually suffices to return
+        `[classification]`.
+        """
+        return [classification]
+
+    def recombine_classification_from_hierarchy(self, hierarchy):
+        """Given a list of parts in the hierarchy, return the classification string.
+
+        For non-hierarchical taxonomies, it usually suffices to return hierarchy[0].
+        """
+        return hierarchy[0]
+
+    def provide_list_context_and_uptodate(self):
+        """Provide data for the context and the uptodate list for the list of all classifiations.
+
+        Must return a tuple of two dicts. The first is merged into the page's context,
+        the second will be put into the uptodate list of all generated tasks.
+
+        Context must contain `title`.
+        """
+        raise NotImplementedError()
+
+    def provide_context_and_uptodate(self, classification):
+        """Provide data for the context and the uptodate list for the list of the given classifiation.
+
+        Must return a tuple of two dicts. The first is merged into the page's context,
+        the second will be put into the uptodate list of all generated tasks.
+
+        Context must contain `title`, which should be something like 'Posts about <classification>',
+        and `classification_title`, which should be related to the classification string.
+        """
+        raise NotImplementedError()
+
+    def postprocess_posts_per_classification(self, posts_per_classification_per_language, flat_hierarchy_per_lang=None, hierarchy_lookup_per_lang=None):
+        """Rearrange, modify or otherwise use the list of posts per classification and per language.
+
+        For compatibility reasons, the list could be stored somewhere else as well.
+
+        In case `has_hierarchy` is `True`, `flat_hierarchy_per_lang` is the flat
+        hierarchy consisting of `TreeNode` elements, and `hierarchy_lookup_per_lang`
```

Comment:
There are lots of features I never used, and category hierarchies are one of them.